### PR TITLE
chore: mark server-related fixture options as such

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: examples/todomvc/playwright.config.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default defineConfig({
   workers: process.env.CI ? 1 : undefined,
 
   /* Reporter to use. See https://playwright.dev/docs/test-reporters */
-  reporter: 'html',
+  reporter: [['html'], ['list']],
```
Comment (dgozman) [inline examples/todomvc/playwright.config.ts:34]: Why this change?

File: packages/playwright/src/runner/testServer.ts [hunk 2]
```diff
@@ -36,17 +38,23 @@ type PlaywrightTestOptions = {
...
-  process.env.FORCE_COLOR = '1';
 
-  const config = await loadConfigFromFile(configFile);
-  if (!config)
-    return;
+  const configFileOrDirectory = configFile ? path.resolve(process.cwd(), configFile) : process.cwd();
+  const resolvedConfigFile = resolveConfigFile(configFileOrDirectory);
+  if (restartWithExperimentalTsEsm(resolvedConfigFile))
+    return null;
+  const configPaths: ConfigPaths = { configFile: resolvedConfigFile, configDir: configFileOrDirectory };
```
Comment (dgozman) [inline packages/playwright/src/runner/testServer.ts:53]:
```suggestion
  const configPaths: ConfigPaths = { configFile: resolvedConfigFile, configDir: resolvedConfigFile ? path.dirname(resolvedConfigFile) : configFileOrDirectory };
```

File: packages/playwright/src/runner/testServer.ts [hunk 3]
```diff
@@ -111,11 +119,12 @@ class Dispatcher {
   }
 
   private async _listTests(reporterPath: string, locations: string[] | undefined) {
-    this._config.cliArgs = [...(locations || []), '--reporter=null'];
-    const reporter = new InternalReporter(new Multiplexer(await createReporters(this._config, 'list', [[reporterPath]])));
-    const taskRunner = createTaskRunnerForList(this._config, reporter, 'out-of-process', { failOnLoadErrors: true });
-    const testRun = new TestRun(this._config, reporter);
-    reporter.onConfigure(this._config.config);
+    const config = await this._loadConfig({});
+    config.cliArgs = [...(locations || []), '--reporter=null'];
```
Comment (dgozman) [inline packages/playwright/src/runner/testServer.ts:123]:
```suggestion
    config.cliArgs = locations || [];
```

File: packages/playwright/src/runner/testServer.ts [hunk 4]
```diff
@@ -129,28 +138,33 @@ class Dispatcher {
...
+
+    const config = await this._loadConfig(overrides);
+    config.cliListOnly = false;
+    config.cliArgs = locations || [];
+    config.cliGrep = options.grep;
+    config.cliProjectFilter = options.projects?.length ? options.projects : undefined;
     // Too late to adjust via overrides for this one.
     if (options.oneWorker)
-      this._config.config.workers = 1;
+      config.config.workers = 1;
```
Comment (dgozman) [inline packages/playwright/src/runner/testServer.ts:162]: Should we remove line 152 then?

File: packages/playwright/src/runner/testServer.ts [hunk 5]
```diff
@@ -111,11 +119,12 @@ class Dispatcher {
...
 
   private async _listTests(reporterPath: string, locations: string[] | undefined) {
-    this._config.cliArgs = [...(locations || []), '--reporter=null'];
-    const reporter = new InternalReporter(new Multiplexer(await createReporters(this._config, 'list', [[reporterPath]])));
-    const taskRunner = createTaskRunnerForList(this._config, reporter, 'out-of-process', { failOnLoadErrors: true });
-    const testRun = new TestRun(this._config, reporter);
-    reporter.onConfigure(this._config.config);
+    const config = await this._loadConfig({});
+    config.cliArgs = [...(locations || []), '--reporter=null'];
+    const reporter = new InternalReporter(new Multiplexer(await createReporters(config, 'list', [[reporterPath]])));
```
Comment (dgozman) [inline packages/playwright/src/runner/testServer.ts:124]: Why not use `additionalReporters` in the same way here?

**Review by dgozman — APPROVED** [review]

File: examples/todomvc/playwright.config.ts (hunk 1 above)
Comment (pavelfeldman) [reply examples/todomvc/playwright.config.ts:34]: I'd like to better test colors in the reporter output.

File: packages/playwright/src/runner/testServer.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright/src/runner/testServer.ts:126]: I do need reporter=null here.

File: packages/playwright/src/runner/testServer.ts (hunk 5 above)
Comment (pavelfeldman) [reply packages/playwright/src/runner/testServer.ts:127]: because they are additional and not overriding.
